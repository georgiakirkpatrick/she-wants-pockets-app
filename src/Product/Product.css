* {
    box-sizing: border-box;
}

.Product {
    width: 46%;
}   

.Product__cropped {
    border-color: black;
    border-width: 2px;
    border-style: solid;
    width: 100%;
    height: 200px;
    object-fit: cover;
    overflow: hidden;
    object-position: 50% 10px ;
}

.Product__cropped img {
    height: 200px;
}