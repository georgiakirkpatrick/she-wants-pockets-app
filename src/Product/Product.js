import React from 'react'
import './Product.css'

function Product(props) {
    return (
        <li className='Product' key={props.id}>
            <div className='Product__cropped'>
                <img src={props.pathToImg} alt={props.imgAlt} />
            </div>
            <h2 className='Product__brand'>{props.brand}</h2>
            <p className='Product__title'>{props.productTitle}</p>
            <p className='Product__price'>{props.price}</p>
        </li>
    )
}

export default Product