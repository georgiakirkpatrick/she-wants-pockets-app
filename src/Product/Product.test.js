import React from 'react';
import ReactDOM from 'react-dom';
import Product from './Product';
import renderer from 'react-test-renderer';

const productProps = {
    id: 1999,
    categoryId: '369',
    pathToImg: 'www.google.com',
    imgAlt: 'Woman in dress',
    brand: 'Sezane',
    productTitle: 'long-sleeved dress',
    price: 98
}

describe ('Product', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<Product />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Product {...productProps}/>)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})