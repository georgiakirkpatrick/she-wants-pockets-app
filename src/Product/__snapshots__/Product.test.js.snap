// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders the UI as expected 1`] = `
<li
  className="Product"
>
  <div
    className="Product__cropped"
  >
    <img
      alt="Woman in dress"
      src="www.google.com"
    />
  </div>
  <h2
    className="Product__brand"
  >
    Sezane
  </h2>
  <p
    className="Product__title"
  >
    long-sleeved dress
  </p>
  <p
    className="Product__price"
  >
    98
  </p>
</li>
`;
